### PR TITLE
Gna split align convert to conv filter

### PR DESCRIPTION
**Pull request by kbruniec**
### Details:
 - Make unaligned split based on Conv instead of Affine
 - Reuse Conv helpers
 - Refactor
 - Enhance GNA model debugging (print tensor Data pointer)

### Tickets:
 - 57319
 - 57317

### CODE REVIEW

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp [hunk 1]
```diff
@@ -1762,68 +1760,76 @@ void GNAGraphCompiler::AffineFilterPrimitive(InferenceEngine::CNNLayerPtr layer)
...
+    //     gnamem->readonly().push_initializer(ptr_weights, paddedWeightsSize, [=](void* data, size_t size) {
+    //         size_t offset = 0;
+    //         for (uint32_t i = 0; i < num_rows_out && size >= offset; i++) {
+    //             ie_memcpy(reinterpret_cast<uint8_t*>(data) + offset, size - offset,
+    //                 filterLayer->_weights->cbuffer().as<const uint8_t*>() + num_rows_in * i * filterLayer->precision.size(),
+    //                 num_rows_in* filterLayer->precision.size());
+    //             offset += (num_rows_in + num_padding) * filterLayer->precision.size();
+    //         }
+    //         }, 64);
+    // }
```
Comment (kbruniec) [inline inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1824]:
```suggestion
```

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp [hunk 2]
```diff
@@ -1762,68 +1760,76 @@ void GNAGraphCompiler::AffineFilterPrimitive(InferenceEngine::CNNLayerPtr layer)
...
 
     connectInput(layer, ptr_inputs, num_data_bytes_in, 0, 0);
     connectOutput(layer, ptr_outputs, num_data_bytes_out);
 
-    if (num_padding == 0) {
+    //if (num_padding == 0) {
         gnamem->readonly().push_ptr(ptr_weights,
             filterLayer->_weights->cbuffer().as<const void*>(),
             filterLayer->_weights->byteSize(),
             64);
```
Comment (kbruniec) [inline inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1809]:
```suggestion
    gnamem->readonly().push_ptr(ptr_weights,
        filterLayer->_weights->cbuffer().as<const void*>(),
        filterLayer->_weights->byteSize(),
        64);
```

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp [hunk 3]
```diff
@@ -162,7 +162,8 @@ void GNAGraphCompiler::fillSplitConnections(InferenceEngine::CNNLayerPtr layer)
                         InferenceEngine::details::product(begin(dataOutput->getDims()),
                                                         end(dataOutput->getDims())) * dataOutput->getPrecision().size();
 
-                if (LayerInfo(outFunctionalLayer.first).isAffineFilter()) {
+                if (//LayerInfo(outFunctionalLayer.first).isAffineFilter() ||
+                    LayerInfo(outFunctionalLayer.first).isConvolutionFilter()) {
```
Comment (kbruniec) [inline inference-engine/src/gna_plugin/gna_graph_compiler.cpp:166]:
```suggestion
                if (LayerInfo(outFunctionalLayer.first).isConvolutionFilter()) {
```

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp [hunk 4]
```diff
@@ -1886,13 +1875,19 @@ void GNAGraphCompiler::PWLPrimitive(InferenceEngine::CNNLayerPtr layer) {
...
         if (rowsCopiedOffset != 0) {
             num_rows -= rowsCopiedOffset / outputs->getPrecision().size();
             layer->params["output_offset"] = std::to_string(rowsCopiedOffset);
         }
+    } else if (LayerInfo(prevLayer).isConvolutionFilter()) {
+        const auto num_rows_for_pwl = prevLayer->GetParamAsInt("num_rows_for_pwl", 0);
+        if (num_rows_for_pwl != 0) {
+            num_rows = num_rows_for_pwl;
+        }
+
```
Comment (kbruniec) [inline inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1890]:
```suggestion
```

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp [hunk 5]
```diff
@@ -1728,8 +1715,8 @@ void GNAGraphCompiler::ConcatAlignFilterPrimitive(InferenceEngine::CNNLayerPtr l
     }
 }
 
-void GNAGraphCompiler::AffineFilterPrimitive(InferenceEngine::CNNLayerPtr layer) {
```
[the remark this answers is not in the document]
Comment (kbruniec) [reply inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1731]: I am not aware of such imported networks.
When we export a model (e.g., using -wg of speech_samle) we use the form which is very close to GNA API form, which is at lower level then the AffineFilter.
In other words: when we import any model there is **no chance** to process it with this deleted function (`AffineFilterPrimitive`).

@nkogteva Did I addressed your concerns?

[the remark this answers is not in the document]
Comment (nkogteva) [reply inference-engine/src/gna_plugin/gna_graph_compiler.cpp:1731]: @kbruniec yes

**Review by nkogteva — APPROVED** [review]

File: inference-engine/src/gna_plugin/backend/am_intel_dnn.cpp [hunk 6]
```diff
@@ -1619,11 +1588,11 @@ void GNAPluginNS::backend::AMIntelDNN::InitGNAStruct(intel_nnet_type_t *ptr_nnet
                     pConvolutionalLayer->nBytesBias = component[i].op.conv1D.num_bytes_per_bias;
                     pConvolutionalLayer->nBytesFilterCoefficient = component[i].op.conv1D.num_bytes_per_weight;
                     pConvolutionalLayer->nFilters = component[i].op.conv1D.num_filters;
-                    pConvolutionalLayer->nFilterRows = comp.op.conv1D.num_filter_coefficients / comp.op.conv1D.num_feature_map_columns;
+                    pConvolutionalLayer->nFilterRows = comp.op.conv1D.num_filter_coefficients / comp.op.conv1D.convStride;
                     pConvolutionalLayer->nFilterCoefficients = component[i].op.conv1D.num_filter_coefficients;
                     pConvolutionalLayer->nFeatureMaps = 1;
-                    pConvolutionalLayer->nFeatureMapRows = component[i].op.conv1D.num_feature_map_rows;
-                    pConvolutionalLayer->nFeatureMapColumns = component[i].op.conv1D.num_feature_map_columns;
+                    pConvolutionalLayer->nFeatureMapColumns = component[i].op.conv1D.convStride;
```
Comment (sirzabek) [inline inference-engine/src/gna_plugin/backend/am_intel_dnn.cpp:1594]: Wouldn't it make sense to change the variables naming here as well? Currently "stride" is used to some point and then again "feature map" is used and it's not obvious what's the connection between stride and "feature map columns". Or am I missing something?

File: inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp [hunk 7]
```diff
@@ -345,6 +345,10 @@ inline void quantizeWeightsBiases(const QuantDesc & quantDesc,
         num_columns = wl->_weights->size() / num_rows;
     }
 
+    if (LayerInfo(wl).isConvolutionFilter()) {
```
Comment (sirzabek) [inline inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp:348]: Maybe some comment here would be useful, why such condition and assignment.

**Review by sirzabek — APPROVED** [review]

File: inference-engine/src/gna_plugin/backend/am_intel_dnn.cpp (hunk 6 above)
Comment (kbruniec) [reply inference-engine/src/gna_plugin/backend/am_intel_dnn.cpp:1594]: These are part of GNA Library 1.0 API - we can not change it.
It is not a big problem as will be removed once the support for GNA 1.0 library not needed.

Comment (sirzabek) [reply inference-engine/src/gna_plugin/backend/am_intel_dnn.cpp:1594]: OK, thanks for explanation.

**Pull request merged**